bluesky/callbacks/fitting.py

```diff
@@ -290,5 +291,33 @@ def compute(self):
             if len(_cen_list) >= 2:
                 self.fwhm = np.abs(self.crossings[-1] - self.crossings[0],
                                    dtype=float)
+
+        # Calculate the derivative of the data
+        x_der = x[1:]
+        y_der = np.diff(y)
+        self.x_der_data = x_der
+        self.y_der_data = y_der
+        self.max_der = x[np.argmax(y_der)], self.y_data[np.argmax(y_der)],
+        self.min_der = x[np.argmin(y_der)], self.y_data[np.argmin(y_der)],
```

Comment:
Argmin and argnax are calculated twice each, that is not efficient. I think it'd make sense to stash the values into temp vars and reuse.

Reply:
2 questions:
-Should the derivative calculation be optional? I.e. only when asked should we spend time computing it, using a flag?
-I am defining the der namedtuple dic in the __init__ instance. Is this the correct way of doing so or should it be only defined in compute?

Reply:
I like the optional approach as it won't break the existing `PeakStats` content.

For the second question, defining it in `__init__` makes sense, and then populate it in the `compute` method.